### PR TITLE
add docker.elastic.co registry support

### PR DESCRIPTION
**Pull request by cristifalcas**
add docker.elastic.co registry support

Based on https://github.com/bazel-contrib/rules_oci/issues/463

### CODE REVIEW

**Comment by thesayyn** [comment]
Can you also add an oci_pull that fetches from this registry for testing purposes? 

oci_pull target goes here; https://github.com/bazel-contrib/rules_oci/blob/main/fetch.bzl
and repo name goes here: https://github.com/bazel-contrib/rules_oci/blob/f191ce2b08556e1d581db3cb92f7207b41aa40b0/oci/tests/BUILD.bazel#L66

**Comment by cristifalcas** [comment]
Very strange for me this error that is thrown by failing tests
Any help? :)

File: oci/private/pull.bzl [hunk 1]
```diff
@@ -34,6 +34,11 @@ _IMAGE_REFERENCE_ATTRS = {
         default = "@oci_auth_config//:standard_authorization_config_path",
         allow_single_file = True,
     ),
+    "docker.elastic.co": {
```
Comment (thesayyn) [inline oci/private/pull.bzl:37]: this is the wrong place to put it. it should go into _WWW_AUTH

Comment (thesayyn) [reply oci/private/pull.bzl:37]: https://github.com/bazel-contrib/rules_oci/blob/f191ce2b08556e1d581db3cb92f7207b41aa40b0/oci/private/pull.bzl#L76

Comment (cristifalcas) [reply oci/private/pull.bzl:37]: face_palm

**Comment by cristifalcas** [comment]
tests pass now on my laptop

**Comment by cristifalcas** [comment]
ping

**Review by thesayyn — APPROVED** [review]

**Pull request merged**